Explanatory Data Analysis
- Clint Atterberry

- Load in your csv.gz's of results for each year extracted.
    1. Concatenate the data into 1 dataframe for the remainder of the analysis.
- Once you have your data from the API, they would like you to perform some light EDA to show:
    1. How many movies had at least some valid financial information (values > 0 for budget OR revenue)?
        - Please exclude any movies with 0's for budget AND revenue from the remaining visualizations.
    2. How many movies are there in each of the certification categories (G/PG/PG-13/R)?
    3. What is the average revenue per certification category?
    4. What is the average budget per certification category?

# Setup

In [33]:
import pandas as pd
import numpy as np


In [6]:
df_2000 = pd.read_csv('Data/final_tmdb_data_2000.csv.gz')
df_2000.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres  \
0         NaN                                                NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   

                               homepage        id original_language  \
0                                   NaN       NaN               NaN   
1                                   NaN   62127.0                en   
2                                   NaN  110977.0                en   
3                                   NaN  442869.0                hi   
4  http://www.wkw-inthemoodforlove.com/     843.0                cn   

    original_title  ...     revenue  runtime  \
0              NaN  ...         NaN      NaN   
1  The Fantasticks  ...         0.0     86.0   
2    For the Cause  ...         0.0    100.0   
3             Gang  ...         0.0    152.0   
4             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         0.000        0.0           NaN  
4    0.0         8.106     1952.0            PG  

[5 rows x 26 columns]

In [7]:
df_2001 = pd.read_csv('Data/final_tmdb_data_2001.csv.gz')
df_2001.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0035423    0.0  /ab5yL8zgRotrICzGbEl10z24N71.jpg                   NaN   
2  tt0114447    0.0                               NaN                   NaN   
3  tt0118589    0.0  /9NZAirJahVilTiDNCHLFcdkwkiy.jpg                   NaN   
4  tt0118652    0.0  /mWxJEFRMvkG4UItYJkRDMgWQ08Y.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  48000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 14, 'nam...      NaN   
2         0.0  [{'id': 53, 'name': 'Thriller'}, {'id': 28, 'n...      NaN   
3  22000000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...      NaN   
4   1000000.0  [{'id': 27, 'name': 'Horror'}, {'id': 9648, 'n...      NaN   

         id original_language         original_title  ...     revenue  \
0       NaN               NaN                    NaN  ...         NaN   
1   11232.0                en         Kate & Leopold  ...  76019048.0   
2  151007.0                en       The Silent Force  ...         0.0   
3   10696.0                en                Glitter  ...   5271666.0   
4   17140.0                en  The Attic Expeditions  ...         0.0   

   runtime                                   spoken_languages    status  \
0      NaN                                                NaN       NaN   
1    118.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2     90.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3    104.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
4    100.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline                  title  \
0                                                NaN                    NaN   
1  If they lived in the same century they'd be pe...         Kate & Leopold   
2  They left him for dead... They should have fin...       The Silent Force   
3   In music she found her dream, her love, herself.                Glitter   
4  His search for peace of mind... will leave his...  The Attic Expeditions   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0           6.3     1137.0         PG-13  
2    0.0           5.0        3.0           NaN  
3    0.0           4.6      117.0         PG-13  
4    0.0           5.1       28.0             R  

[5 rows x 26 columns]

In [8]:
# Create 1 df from both dfs
df = pd.concat([df_2000,df_2001])
df.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres  \
0         NaN                                                NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   

                               homepage        id original_language  \
0                                   NaN       NaN               NaN   
1                                   NaN   62127.0                en   
2                                   NaN  110977.0                en   
3                                   NaN  442869.0                hi   
4  http://www.wkw-inthemoodforlove.com/     843.0                cn   

    original_title  ...     revenue  runtime  \
0              NaN  ...         NaN      NaN   
1  The Fantasticks  ...         0.0     86.0   
2    For the Cause  ...         0.0    100.0   
3             Gang  ...         0.0    152.0   
4             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         0.000        0.0           NaN  
4    0.0         8.106     1952.0            PG  

[5 rows x 26 columns]

# Analysis

In [18]:
finance_filter = df[(df['budget'] > 0) & (df['revenue'] > 0)]
finance_filter.count()

imdb_id                  349
adult                    349
backdrop_path            337
belongs_to_collection     84
budget                   349
genres                   349
homepage                  56
id                       349
original_language        349
original_title           349
overview                 349
popularity               349
poster_path              349
production_companies     349
production_countries     349
release_date             349
revenue                  349
runtime                  349
spoken_languages         349
status                   349
tagline                  316
title                    349
video                    349
vote_average             349
vote_count               349
certification            305
dtype: int64

## How many movies had at least some valid financial information (values > 0 for budget OR revenue)?

There appears to be 349 results that have their budget or revenue data.

In [21]:
df = finance_filter
df.head()

imdb_id  adult                     backdrop_path  \
4   tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
11  tt0120467    0.0  /knok3mNReKqPTplnnqz7E4dd7mD.jpg   
13  tt0120630    0.0  /sPAwM8WxMdXNlqeP4F9DMe9LYoY.jpg   
16  tt0120753    0.0                               NaN   
17  tt0120755    0.0  /24DZfupDlhXeTchmcOkoGRhP5Vg.jpg   

                                belongs_to_collection       budget  \
4                                                 NaN     150000.0   
11                                                NaN     120000.0   
13  {'id': 718551, 'name': 'Chicken Run Collection...   45000000.0   
16                                                NaN    8000000.0   
17  {'id': 87359, 'name': 'Mission: Impossible Col...  125000000.0   

                                               genres  \
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
11  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...   
13  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
16  [{'id': 18, 'name': 'Drama'}, {'id': 53, 'name...   
17  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   

                                homepage       id original_language  \
4   http://www.wkw-inthemoodforlove.com/    843.0                cn   
11                                   NaN  19085.0                en   
13                                   NaN   7443.0                en   
16                                   NaN    318.0                en   
17                                   NaN    955.0                en   

              original_title  ...      revenue  runtime  \
4                       花樣年華  ...   12854953.0     99.0   
11                    Vulgar  ...      14904.0     87.0   
13               Chicken Run  ...  224834564.0     84.0   
16  The Million Dollar Hotel  ...     105983.0    122.0   
17    Mission: Impossible II  ...  546388105.0    123.0   

                                     spoken_languages    status  \
4   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
11  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
13  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
16  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
17  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                              tagline  \
4   Feel the heat, keep the feeling burning, let t...   
11   Everyone loves a clown... some more than others.   
13  This ain't no chick flick. It's poultry in mot...   
16                    Everyone has something to hide.   
17                       Expect the impossible again.   

                       title  video  vote_average vote_count certification  
4       In the Mood for Love    0.0         8.106     1952.0            PG  
11                    Vulgar    0.0         5.400       42.0             R  
13               Chicken Run    0.0         6.725     4104.0             G  
16  The Million Dollar Hotel    0.0         5.915      254.0             R  
17    Mission: Impossible II    0.0         6.107     5444.0         PG-13  

[5 rows x 26 columns]

In [27]:
# Count of movies in each category
df.certification.value_counts()

R        140
PG-13    117
PG        29
G         13
NR         6
Name: certification, dtype: int64

In [55]:
# mean revenue by category
cert_filter = df.groupby(['certification']).mean()
cert_filter.revenue.round()

certification
G        133216882.0
NR        23823316.0
PG       134613828.0
PG-13    110712060.0
R         53446539.0
Name: revenue, dtype: float64

In [56]:
# mean budget by category
cert_filter = df.groupby(['certification']).mean()
cert_filter.budget.round()

certification
G        44000000.0
NR       14783333.0
PG       53741380.0
PG-13    46806815.0
R        27502554.0
Name: budget, dtype: float64

# Saving file to CSV.GZ

In [59]:
df.to_csv('Data/tmdb_results_combined.csv.gz', compression='gzip', index=False)